ssl/statem/statem_clnt.c

```diff
@@ -1220,6 +1220,7 @@ __owur CON_FUNC_RETURN tls_construct_client_hello(SSL_CONNECTION *s,
     /* Work out what SSL/TLS/DTLS version to use */
     int protverr = ssl_set_client_hello_version(s);
 
+    memset(&inner, 0, sizeof(inner));
```

Comment:
I do not know if this pattern of memsetting the WPACKET to be able to call WPACKET_cleanup() unconditionally is used elsewhere.

Reply:
Fair enough. I'll take a peek at this again tomorrow when back at things. (Incl. your comment about not needing `BUF_MEM_grow()`.